web/src/engine/main/src/engineConfiguration.ts

```diff
@@ -0,0 +1,108 @@
+import { DeviceSpec, ManagedPromise, physicalKeyDeviceAlias, SpacebarText } from "@keymanapp/keyboard-processor";
+import { PathConfiguration, PathOptionDefaults, PathOptionSpec } from "keyman/engine/configuration";
+import { Device } from "keyman/engine/device-detect";
+import { KeyboardStub } from "keyman/engine/keyboard-cache";
+
+export class EngineConfiguration {
+  // The app/webview path replaces this during init, but we expect to have something set for this
+  // during engine construction, which occurs earlier.  So no `readonly`, sadly.
+  //
+  // May also be manipulated by Developer's debug-host?
+  public hostDevice: DeviceSpec;
+  readonly sourcePath: string;
+  readonly deferForInitialization: ManagedPromise<void>;
+
+  private _paths: PathConfiguration;
+  private _activateFirstKeyboard: boolean;
+  private _defaultSpacebarText: SpacebarText;
+  private _stubNamespacer?: (KeyboardStub) => void;
+
+  public applyCacheBusting: boolean = false;
+
+  // sourcePath:  see `var sPath =` in kmwbase.ts.  It is not obtainable headlessly.
+  constructor(sourcePath: string, device?: DeviceSpec) {
+    if(!device) {
+      const deviceDetector = new Device();
+      deviceDetector.detect();
+
+      device = deviceDetector.coreSpec;
+    }
+
+    this.sourcePath = sourcePath;
+    this.hostDevice = device;
+    this.deferForInitialization = new ManagedPromise<void>();
+  }
+
+  initialize(options: Required<InitOptionSpec>) {
+    this._paths = new PathConfiguration(options, this.sourcePath);
+    if(typeof options.setActiveOnRegister == 'boolean') {
+      this._activateFirstKeyboard = options.setActiveOnRegister;
+    } else if (typeof options.setActiveOnRegister == 'string') {
+      let str = options.setActiveOnRegister.toLowerCase();
+      this._activateFirstKeyboard = str === 'true';
+    } else {
+      this._activateFirstKeyboard = true;
+    }
+
+    this._defaultSpacebarText = options.spacebarText;
+
+    this.deferForInitialization.resolve();
+  }
+
+  get paths() {
+    return this._paths;
+  }
+
+  get activateFirstKeyboard() {
+    return this._activateFirstKeyboard;
+  }
+
+  get defaultSpacebarText() {
+    return this._defaultSpacebarText;
+  }
+
+  get softDevice(): DeviceSpec {
+    return this.hostDevice;
+  }
+
+  get hardDevice(): DeviceSpec {
+    return physicalKeyDeviceAlias(this.hostDevice);
+  }
+
+  get stubNamespacer() {
+    return this._stubNamespacer;
+  }
+
+  set stubNamespacer(functor: (stub: KeyboardStub) => void) {
+    this._stubNamespacer = functor;
+  }
+
+  debugReport(): Record<string, any> {
```

Comment:
Intended for use with `keyman.getDebugInfo()`.

_As seen in the Sentry dashboard:_
![image](https://user-images.githubusercontent.com/25213402/225809608-8c3835e4-94a6-45b8-a875-50e2ffd799d2.png)

At this point, it may well implement most of the current report seen above, but we will likely add a few extra properties from other sources (like the engine object and/or the OSK) as well - so this isn't intended as the top-level call for said function.